### PR TITLE
#5346: [TEST] Collect application logs if test is failed

### PR DESCRIPTION
**Pull request by pkazlenka**
Implements #5346

* Creates json file with kilda application logs in build/logs/spec/<specgroup>/<specname>/ directory
if test is failed.

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    private static String utcTimeNow() {
+        return now(ZoneOffset.UTC).format(ISO_DATE_TIME)
+    }
+
+    private static File getTargetLogFile(ErrorInfo error) {
+        def specGroupName = error.getMethod().getParent().getPackage().split('\\.').last()
+        def specName = error.getMethod().getParent().getName()
+        def iterationName = error.getMethod().getIteration().getDisplayName()
+        return new File("build/logs/spec/${specGroupName}/${specName}/${iterationName}.server.log.json")
```
Comment (pkazlenka) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy:53]: Please, let me know it there's a way to retrieve gradle's build directory root in runtime.

Comment (dmitrii-beliakov) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy:50]: Not sure how to use gradle API from here, but you can save something as a system property in gradle.build and then read it here. For example, in a random test I had opened it looks like this:
```java
getClass().getClassLoader().getResourceAsStream("topology.properties")
```
then if you write sources root to a property, then you can get it here back.
But maybe this is overkill for this case.

**Review by dmitrii-beliakov — APPROVED** [review]

**Review by rtretyak — APPROVED** [review]

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                        "@timestamp:[${startTime} TO ${utcTimeNow()}]", String.class).getBody()
+        def beautifiedString = objectMapper.readValue(logs, Object.class)
+        objectMapper.writerWithDefaultPrettyPrinter().writeValue(getTargetLogFile(error), beautifiedString)
+    }
+
+    private static String utcTimeNow() {
+        return now(ZoneOffset.UTC).format(ISO_DATE_TIME)
+    }
+
+    private static File getTargetLogFile(ErrorInfo error) {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy:44]: It'd be a good idea for this and https://github.com/telstra/open-kilda/blob/558e34f3f9476aa0f83b024657ff982d862271dc/src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/LogParallelSpecsListener.groovy#L43
to find 'logs directory' for current test using some common shared method

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy (hunk 1 above)
Comment (pkazlenka) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy]: Hardcoding would be easier, I'm afraid.

**Pull request merged**